rtl/eclrtl/eclrtl.cpp

```diff
@@ -4851,12 +4852,13 @@ class CStrRegExprFindInstance : implements IStrRegExprFindInstance
         {
             if (_keep)
             {
-                sample = (char *)rtlMalloc(_len + 1);  //required for findstr
+                sample = (char *)rtlMalloc(_len + 1);
                 memcpy(sample, _str + _from, _len);
-                sample[_len] = (char)NULL;
+                sample_end = sample+_len;
+                *sample_end = (char)NULL;
                 matched = boost::regex_search(sample, subs, *regEx);
             }
-            else 
+            else
             {
                 matched = boost::regex_search(_str + _from, _str + _len, subs, *regEx);
```

Comment:
sample_end should also be initialised here - otherwise passing keep = false causes the code to crash.